src/actions/vis-state-actions.js

```diff
@@ -508,6 +520,75 @@ export function setFeatures(features) {
   }
 }
 
+/**
+ * It will apply the provide feature as filter to the given layer.
+ * If the given feature is already applied as filter to the layer it  will remove it from  the filter list
```

Comment:
If the given feature is already applied as filter to the layer, it will __remove the layer__ from the filter